### PR TITLE
Jerky searchbar fix

### PR DESCRIPTION
**Pull request by lekkas**
Bug:
bug

After fix:
fix

### CODE REVIEW

File: static/scripts/main.js [hunk 1]
```diff
@@ -6,23 +6,39 @@ window.onload = function () {
...
+var UnpinOffset = 400;
+var PrevScrollTop = 0;
 
 $(window).scroll(function() {
-  var NextScroll = $(this).scrollTop();
-  if (NextScroll < CurrentScroll && NextScroll > 67){
-     //upscroll
-     $('[data-md-sticky-header]').addClass('sticky');
+  function handleScrollUp() {
+    if (ScrollTop > SearchbarTop && ScrollTop <= UnpinOffset)
```
Comment (jviotti) [inline static/scripts/main.js]: Can you add curly braces around `if`/`else` bodies for safety and consistency?

File: static/scripts/main.js [hunk 2]
```diff
@@ -6,23 +6,39 @@ window.onload = function () {
...
       hash.scrollIntoView()
     }
+    $('[data-md-sticky-header]').headroom({
+      offset: UnpinOffset,
+      tolerance: 0
+    });
 }
 
-var CurrentScroll = 0;
+var SearchbarTop = 61;
```
Comment (jviotti) [inline static/scripts/main.js]: What does this magic number stands for? The height of the header? Can it be computed dynamically?

File: static/scripts/main.js [hunk 3]
```diff
@@ -6,23 +6,39 @@ window.onload = function () {
...
+var PrevScrollTop = 0;
 
 $(window).scroll(function() {
-  var NextScroll = $(this).scrollTop();
-  if (NextScroll < CurrentScroll && NextScroll > 67){
-     //upscroll
-     $('[data-md-sticky-header]').addClass('sticky');
+  function handleScrollUp() {
+    if (ScrollTop > SearchbarTop && ScrollTop <= UnpinOffset)
+      $('[data-md-sticky-header]').addClass('sticky');
```
Comment (jviotti) [inline static/scripts/main.js]: This element is being queried on each scroll event. Can we save a reference outside the scroll event handler and reuse here?

**Comment by jviotti** [comment]
Looks really nice!

File: static/scripts/main.js [hunk 4]
```diff
@@ -6,23 +6,39 @@ window.onload = function () {
...
     }
+    $('[data-md-sticky-header]').headroom({
+      offset: UnpinOffset,
+      tolerance: 0
+    });
 }
 
-var CurrentScroll = 0;
+var SearchbarTop = 61;
+var UnpinOffset = 400;
```
Comment (jviotti) [inline static/scripts/main.js]: This variable is used when applying `headroom` on the header at line 9, however the variable is declared at line 15, which could cause some confusion. Can we move this variable to a place before it's used?

File: static/scripts/main.js (hunk 1 above)
Comment (emirotin) [inline static/scripts/main.js]: yeah, even single-statement blocks should be in braces

this code (I mean the entire source, not this PR specifically) actually needs linting, for example, variables should start from lower-case letter

**Comment by emirotin** [comment]
Cool improvement, just code style nits

File: static/scripts/main.js (hunk 2 above)
Comment (lekkas) [inline static/scripts/main.js]: It is the topmost vertical position of the searchbar. I agree we shouldn't be using a magic value for that, we can compute it dynamically. I'll look into that, I think $(".search-wrapper").offset().top will do.

File: static/scripts/main.js (hunk 1 above)
Comment (lekkas) [inline static/scripts/main.js]: Fair point, will add curly braces in the next commit.

File: static/scripts/main.js (hunk 3 above)
Comment (lekkas) [inline static/scripts/main.js]: :+1:

**Comment by jviotti** [comment]
LGTM

**Pull request merged**